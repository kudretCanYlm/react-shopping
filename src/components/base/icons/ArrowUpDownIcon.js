import { BsFillArrowUpCircleFill } from 'react-icons/bs';

const ArrowUpDownIcon = ({ fontSize = 20, className = '' }) => {
  return (
    <BsFillArrowUpCircleFill fontSize={fontSize} className={`arrow-up-down-icon ${className}`} />
  );
};

export default ArrowUpDownIcon;
