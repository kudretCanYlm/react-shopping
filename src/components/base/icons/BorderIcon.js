import React from 'react';
import PropTypes from 'prop-types';
import { EMPTY_ICON } from 'components/base/links/common/common-props';

const BorderIcon = ({ children, onClick = (f) => f, className }) => {
  return (
    <div onClick={onClick} className={`border-icon flex-row ${className}`}>
      {children}
    </div>
  );
};

//prop-type
BorderIcon.propTypes = {
  children: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string
};

//default props
BorderIcon.defaultProps = {
  children: EMPTY_ICON,
  className: ''
};

export default BorderIcon;
