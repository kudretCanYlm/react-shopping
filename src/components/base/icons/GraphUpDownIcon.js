import { BsGraphUp, BsGraphDown } from 'react-icons/bs';

const GraphUpDownIcon = ({ isToUp = false, className = '', fontSize = 40 }) => {
  return (
    <div className={`graph-up-down-icon ${className}`}>
      {isToUp ? (
        <BsGraphUp className="bs-graph-icon icon-up" fontSize={fontSize} />
      ) : (
        <BsGraphDown className="bs-graph-icon icon-down" fontSize={fontSize} />
      )}
    </div>
  );
};

export default GraphUpDownIcon;
