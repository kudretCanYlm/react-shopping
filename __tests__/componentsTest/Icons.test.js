import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
//components
import { AiFillClockCircle, AiFillApple } from 'react-icons/ai';
import ArrowUpDownIcon from 'components/base/icons/ArrowUpDownIcon';
import BorderIcon from 'components/base/icons/BorderIcon';
import GraphUpDownIcon from 'components/base/icons/GraphUpDownIcon';

Enzyme.configure({ adapter: new Adapter() });

//varibles
const iconClassName = 'test className';
const iconfontSize = 20;
var textBefore = 'test';
const textAfter = 'test clicked';

describe('ArrowUpDownIconTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(<ArrowUpDownIcon className={iconClassName} fontSize={iconfontSize} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('it has my className and fontsize', () => {
    const iconCom = shallow(<ArrowUpDownIcon className={iconClassName} fontSize={iconfontSize} />);

    expect(iconCom.html()).toContain(`font-size=\"${iconfontSize}\"`);
    expect(iconCom.hasClass(iconClassName)).toEqual(true);

    expect(iconCom.html()).not.toContain(`font-size=\"40\"`);
    expect(iconCom.hasClass('not include ClassName')).toEqual(false);
  });
});

describe('BorderIconTest', () => {
  const children = <AiFillClockCircle fontSize={20} />;
  const notChildren = <AiFillApple fontSize={20} />;

  const onClick = () => {
    textBefore = textBefore + ' clicked';
  };

  it('render correctly and work my onClick function', () => {
    const component = renderer
      .create(
        <BorderIcon onClick={onClick} className={iconClassName}>
          {children}
        </BorderIcon>
      )
      .toJSON();

    renderer.act(() => {
      component.props.onClick();
    });

    expect(component).toMatchSnapshot();
    expect(textBefore).toEqual(textAfter);
  });

  it('it has my className and my children', () => {
    const iconCom = shallow(
      <BorderIcon onClick={onClick} className={iconClassName}>
        {children}
      </BorderIcon>
    );

    expect(iconCom.hasClass(iconClassName)).toEqual(true);
    expect(iconCom.containsMatchingElement(children)).toEqual(true);

    expect(iconCom.hasClass('not include ClassName')).toEqual(false);
    expect(iconCom.containsMatchingElement(notChildren)).toEqual(false);
  });
});

describe('GraphUpDownIconTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(<GraphUpDownIcon className={iconClassName} fontSize={iconfontSize} isToUp />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('it has my className and my fontSize', () => {
    const iconCom = shallow(
      <GraphUpDownIcon className={iconClassName} fontSize={iconfontSize} isToUp />
    );

    expect(iconCom.hasClass(iconClassName)).toEqual(true);
    expect(iconCom.html()).toContain(`font-size=\"${iconfontSize}\"`);

    expect(iconCom.hasClass('not include ClassName')).toEqual(false);
    expect(iconCom.html()).not.toContain(`font-size=\"40\"`);
  });

  it('istoUp=true working', () => {
    const iconCom = shallow(
      <GraphUpDownIcon className={iconClassName} fontSize={iconfontSize} isToUp={true} />
    );

    expect(iconCom.children().hasClass('icon-up')).toEqual(true);
    expect(iconCom.children().hasClass('icon-down')).toEqual(false);
  });

  it('isToUp=false working', () => {
    const iconCom = shallow(
      <GraphUpDownIcon className={iconClassName} fontSize={iconfontSize} isToUp={false} />
    );

    expect(iconCom.children().hasClass('icon-down')).toEqual(true);
    expect(iconCom.children().hasClass('icon-up')).toEqual(false);
  });
});
