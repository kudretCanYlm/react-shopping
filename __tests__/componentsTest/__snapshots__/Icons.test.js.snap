// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowUpDownIconTest render correctly 1`] = `
<svg
  className="arrow-up-down-icon test className"
  fill="currentColor"
  fontSize={20}
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 16 16"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M16 8A8 8 0 1 0 0 8a8 8 0 0 0 16 0zm-7.5 3.5a.5.5 0 0 1-1 0V5.707L5.354 7.854a.5.5 0 1 1-.708-.708l3-3a.5.5 0 0 1 .708 0l3 3a.5.5 0 0 1-.708.708L8.5 5.707V11.5z"
  />
</svg>
`;

exports[`BorderIconTest render correctly and work my onClick function 1`] = `
<div
  className="border-icon flex-row test className"
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    fontSize={20}
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm176.5 585.7l-28.6 39a7.99 7.99 0 0 1-11.2 1.7L483.3 569.8a7.92 7.92 0 0 1-3.3-6.5V288c0-4.4 3.6-8 8-8h48.1c4.4 0 8 3.6 8 8v247.5l142.6 103.1c3.6 2.5 4.4 7.5 1.8 11.1z"
    />
  </svg>
</div>
`;

exports[`GraphUpDownIconTest render correctly 1`] = `
<div
  className="graph-up-down-icon test className"
>
  <svg
    className="bs-graph-icon icon-up"
    fill="currentColor"
    fontSize={20}
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h1v15h15v1H0V0zm10 3.5a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-1 0V4.9l-3.613 4.417a.5.5 0 0 1-.74.037L7.06 6.767l-3.656 5.027a.5.5 0 0 1-.808-.588l4-5.5a.5.5 0 0 1 .758-.06l2.609 2.61L13.445 4H10.5a.5.5 0 0 1-.5-.5z"
      fillRule="evenodd"
    />
  </svg>
</div>
`;
